beps/0001-notifications-system/README.md

```diff
@@ -0,0 +1,255 @@
+
+### Notifications Plugin
+
+The role of the notifications plugin is to manage the lifecycle of notifications. The backend plugin provides an API for other backends to send notifications, as well as an accompanying [backend service](https://backstage.io/docs/backend-system/architecture/services). It also provides a separate API for the frontend plugin to read notifications for an individual user and manage the read status of notifications.
+
+The notification backend stores notification using the [database service](https://backstage.io/docs/backend-system/core-services/index#database). In particular it needs to store the following information for each notification:
+
+- ID
+- Payload
+- Receivers
+- Read status by user
+
+The receivers is **not** a list of users, but rather a filter that describes who should receive the notification. It must be possible to evaluate this filter in a database query, so that we can efficiently fetch all notifications for a given user. The same filter will also be used by the signal backend to determine which users should receive a signal.
+
+The notification backend does not provide any new permissions, since creating notifications can only be done by other backend plugins, while reading notifications can only be done by the authenticated user. It is possible that we want to add a permissions for reading notifications, in particular for admin and impersonation use cases, but that is not part of this proposal or the initial implementation.
+
+The notification frontend plugin provides a UI for viewing notifications, which in the initial implementation can be as simple as needed. The only requirement is that a user is able to view recent notifications and distinguish between read and unread notifications. A notification as marked as read once it has been interacted with. The frontend plugin also subscribes to the notifications signal channel and alerts the user when a new notification is received.
+
+### Architecture Overview
+
+The diagram below show an overview of the notifications system architecture and the interaction between the different components.
+
+![notifications system architecture diagram](./notifications-architecture.drawio.svg)
+
+The components that are presented with dashed lines are not directly part of this proposal, but included to be considered as future extensions.
+
+Let's walk through the process of sending of a notification to a user:
+
+1. A backend plugin that wants to send a notification to a user uses the `NotificationService`, which in turn makes a POST request to the notification backend. The request contains a filter to select the target users, as well as the notification payload.
```

Comment:
What kind of filter is this? CatalogFilter or something similar? Implementation details IMO, but something that could be perhaps thought through

Reply:
TBD in the details section